Update documents on building Windows agent (#6741)
DataDog/datadog-agent: docs/dev/agent_omnibus.md

For Linux, we provide Docker images (one to build DEB packages and one for RPM),
with the build dependencies installed, so you don't have to install them on your system.

> ## Building inside Linux Docker image (Linux host only, recommended)

Reviewer: ```suggestion
## Linux Docker image (Linux host only, recommended)
```

itself: running the task multiple times would recursively add those artifacts to
the source files for the `datadog-agent` software definition.

> ## Building inside Windows Docker image (Windows host only)

Reviewer 2: ```suggestion
## Building inside Windows Docker image (Windows host only, recommended)
```

Author: There's no other documented way to build on Windows, I don't think we need to say it's "recommended"

Author: Fixed in latest commit

> ## Building inside Windows Docker image (Windows host only)

Reviewer: ```suggestion
## Windows Docker image (Windows host only, recommended)
```

### Prerequisites

> To build on Windows, [Docker Desktop](https://docs.docker.com/docker-for-windows/install/) must be installed and configured to use Windows container.

Reviewer 2: ```suggestion
To build on Windows, [Docker Desktop](https://docs.docker.com/docker-for-windows/install/) must be installed and configured to use Windows containers.
```

> Start a Command Prompt with Administrators permission and navigate to the local clone of the dataog-agent repo, star Docker image by running:

Reviewer 2: ```suggestion
Start a Command Prompt with Administrators permission and navigate to the local clone of the dataog-agent repo, start Docker image by running:
```
Are admin permissions really needed? Why?

Author: Docker namepipe is ACL'd to Administrators group by default.

> Start a Command Prompt with Administrators permission and navigate to the local clone of the dataog-agent repo, star Docker image by running:

Reviewer 3: ```suggestion
Start a Command Prompt with Administrators permission and navigate to the local clone of the datadog-agent repo, star Docker image by running:
```

Author: Thanks. Fixed in latest commit.

> Start a Command Prompt with Administrators permission and navigate to the local clone of the dataog-agent repo, start Docker image by running:

Reviewer: ```suggestion
Start a command prompt with administrators permission and navigate to your local clone of the `dataog-agent` repo. Start Docker image by running:
```

```

> docker run -v %CD%:c:\dev\go\src\github.com\DataDog\datadog-agent -it datadog/agent-buildimages-windows_1809_x64 cmd

Reviewer 2: Is 1809 going to work for all users, or some will need the 1909 version?

Author: This is the Windows version within docker image, with Hyper-V isolation, this version shouldn't matter.

Author: 1809 is the long term servicing version, we should document this one to avoid changing image version when non lTSC becomes unsupported.

```

> It will take a while on the first run.

Reviewer 2: ```suggestion
It will take a while on the first run since it has to download the ~12GB Docker image.
```

> It will take a while on the first run.

Reviewer: ```suggestion
It takes a while on the first run since it has to download the ~12GB Docker image.
```

Author: Fixed in latest commit

Author: Thanks. Fixed in latest commit.

Author: Thanks. Fixed in latest commit.
